### PR TITLE
rewriting some tests using new limsmock

### PR DESCRIPTION
**Pull request by mayabrandi**
### This PR rewrites some tests using the new limsmock package
- 

**How to test**:
- [ ] clone and run `pytest tests/get/test_artifacts.py`

### Review:
- [ ] Code approved by
- [ ] Tests executed by
- [ ] "Merge and deploy" approved by

This [version](https://semver.org/) is a:
- [ ] **MAJOR** - when you make incompatible API changes
- [x] **MINOR** - when you add functionality in a backwards compatible manner
- [ ] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

### CODE REVIEW

File: tests/EPPs/test_make_kapa_csv.py [hunk 1]
```diff
@@ -1,37 +1,27 @@
...
-    # GIVEN: A file name. and a lims with three samples that has been run
+    server_make_kapa_csv, kapa_library_preparation_file):
+    # GIVEN: A file name. and a lims with two samples that has been run
     # through a process of type <amount_step>. There  they got udf: Amount needed (ng) set.
-    # The tree samples are now in another step where they have location and
+    # The two samples are now in another step where they have location and
     # reagent labels set.
     # (See entety_json_data for details)
 
+    lims = Lims("http://127.0.0.1:8000", 'dummy', 'dummy')
```
Comment (Mropat) [inline tests/EPPs/test_make_kapa_csv.py:20]: can this be moved into a fixture? You will be using this IP:port multiple times!

File: tests/EPPs/test_make_kapa_csv.py [hunk 2]
```diff
@@ -1,37 +1,27 @@
...
-            "outputs": kapa_csv_data["amount_artifacts"],
-        },
-    )
-    curent_step_artifacts = helpers.ensure_lims_artifacts(
-        lims, kapa_csv_data["artifacts_kapa_library_preparation"]
-    )
-
+    amount_step = "Aliquot samples for enzymatic fragmentation TWIST v2"
     file = Path(file_name)
+    curent_step_artifacts = [Artifact(lims, id='2-1155129'), Artifact(lims, id='2-1155130')]
```
Comment (Mropat) [inline tests/EPPs/test_make_kapa_csv.py:24]: could the step artifacts also be predefined in fixture?

File: tests/EPPs/test_make_kapa_csv.py (hunk 1 above)
Comment (Mropat) [reply tests/EPPs/test_make_kapa_csv.py:20]: Also for the amount_step variable, its repeated a bunch in tests!

File: tests/EPPs/test_make_kapa_csv.py (hunk 2 above)
Comment (mayabrandi) [reply tests/EPPs/test_make_kapa_csv.py:22]: Im not sure. Maybe we can talk about it and conclude together?

File: tests/EPPs/test_make_kapa_csv.py [hunk 3]
```diff
@@ -44,29 +32,19 @@ def test_get_file_data_and_write_KAPA_library_preparation(
...
-            "outputs": kapa_csv_data["amount_artifacts"],
-        },
-    )
-    curent_step_artifacts = helpers.ensure_lims_artifacts(
-        lims, kapa_csv_data["artifacts_enzymatic_feragmentation"]
-    )
-
+    amount_step = "Aliquot samples for enzymatic fragmentation TWIST v2"
     file = Path(file_name)
+    curent_step_artifacts = [Artifact(lims, id='2-1155124'), Artifact(lims, id='2-1155125')]
```
Comment (moonso) [inline tests/EPPs/test_make_kapa_csv.py:45]: Fixture from previous test can be reused here

File: tests/EPPs/test_make_kapa_csv.py [hunk 4]
```diff
@@ -79,32 +57,20 @@ def test_get_file_data_and_write_Enzymatic_fragmentation(
 
 
 def test_get_file_data_and_write_missing_udf(
-    lims, kapa_library_preparation_file_missing_udf, helpers, kapa_csv_data
-):
-    # GIVEN: A file name. and a lims with three samples that has been run
-    # through a process of type <amount_step>. There some samples did not get the udf: Amount needed (ng) set.
-    # The tree samples are now in another step where they have location and
+        server_make_kapa_csv_missing_udfs, kapa_library_preparation_file_missing_udf, lims):
```
Comment (moonso) [inline tests/EPPs/test_make_kapa_csv.py:60]: Adding type hints makes also the tests much easier to read!
I think my type hints suggestions are wrong but just exemplifying 

```suggestion
        server_make_kapa_csv_missing_udfs: str, kapa_library_preparation_file_missing_udf: str, lims: Lims):
```

File: tests/EPPs/test_twist_qc_amount.py [hunk 5]
```diff
@@ -1,83 +1,74 @@
...
-            "udf": {"Volume (ul)": 10, "Concentration": 10},
-        },
-        {
-            "samples": [{"udf": {"Source": "blood"}}],
-            "udf": {"Volume (ul)": 50, "Concentration": 50},
-        },
-    ]
-
-    artifacts = helpers.ensure_lims_artifacts(lims=lims, artifacts_data=artifacts_data,)
+    a1 = Artifact(lims, id='1')
```
Comment (moonso) [inline tests/EPPs/test_twist_qc_amount.py:11]: Suggest to split these into individual tests that can be called something like
`test_qc_fails_when_concentration_300`

Comment (moonso) [reply tests/EPPs/test_twist_qc_amount.py:11]: It is easier to read and understand why tests fails if one test only test one thing

File: tests/conftest.py [hunk 6]
```diff
@@ -1,56 +1,63 @@
...
+PORT = 8000
+HOST = '127.0.0.1'
 
 @pytest.fixture
-def kapa_csv_data():
-    """Get file path to make_kapa_csv_data.json"""
-
-    with open("tests/fixtures/make_kapa_csv_data.json") as file:
-        return json.load(file)
+def server_flat_tests():
```
Comment (moonso) [inline tests/conftest.py:15]: Could we add a doc string? I can not figure out what this is used for...
Nothing is returned so it is kind of hard to follow

File: tests/conftest.py [hunk 7]
```diff
@@ -1,56 +1,63 @@
...
+    file_path = f"tests/fixtures/make_kapa_csv_missing_udfs"
+    thread = threading.Thread(target=run_server, args=(file_path, HOST, PORT,))
+    thread.daemon = True
+    thread.start()
+    time.sleep(0.1)
 
 @pytest.fixture
-def process():
-    return MockProcess()
+def lims():
```
Comment (moonso) [inline tests/conftest.py:50]:
```suggestion
def lims() -> Lims:
```

File: tests/conftest.py [hunk 8]
```diff
@@ -1,56 +1,63 @@
...
+    """Get genologics lims instance"""
+    lims = Lims(f"http://{HOST}:{PORT}", 'dummy', 'dummy')
+    return lims
```
Comment (moonso) [inline tests/conftest.py:53]:
```suggestion
    return Lims(f"http://{HOST}:{PORT}", 'dummy', 'dummy')
```

File: tests/conftest.py [hunk 9]
```diff
@@ -1,56 +1,63 @@
...
 
 
-@pytest.fixture(name="helpers")
-def fixture_helpers():
-    """Return a class with small helper functions"""
-    return Helpers()
+@pytest.fixture
+def config():
```
Comment (moonso) [inline tests/conftest.py:57]: Suggest to always use Path objects when working with anything on the file system
```suggestion
def config() -> Path:
```

File: tests/conftest.py [hunk 10]
```diff
@@ -1,56 +1,63 @@
...
+    """Get file path to config"""
+
+    return "tests/fixtures/config.yaml"
```
Comment (moonso) [inline tests/conftest.py:60]:
```suggestion
    return Path("tests/fixtures/config.yaml")
```

File: tests/fixtures/config.yaml [hunk 11]
```diff
@@ -1,4 +1,5 @@
 BASEURI: somebase.scilifelab.se
 HOST: somehoast.scilifelab.se
 USERNAME: someuser
-PASSWORD: somepassword
\ No newline at end of file
+PASSWORD: somepassword
+CG_URL: https://something
```
Comment (moonso) [inline tests/fixtures/config.yaml:5]:
```suggestion
CG_URL: https://something

```

File: tests/fixtures/flat_tests/artifacts/4.xml [hunk 12]
```diff
@@ -0,0 +1,16 @@
+<art:artifact xmlns:udf="http://genologics.com/ri/userdefined" xmlns:file="http://genologics.com/ri/file" xmlns:art="http://genologics.com/ri/artifact" uri="http://127.0.0.1:8000/api/v2/artifacts/2-1154123?state=869639" limsid="2-1154123">
```
Comment (moonso) [inline tests/fixtures/flat_tests/artifacts/4.xml:1]: It wold be nice if these had some name that describes more what they are so one can understand why they are different

**Review by moonso — CHANGES_REQUESTED** [review]
Great job!

File: tests/EPPs/test_twist_qc_amount.py (hunk 5 above)
Comment (mayabrandi) [reply tests/EPPs/test_twist_qc_amount.py]: Good point

File: tests/EPPs/test_make_kapa_csv.py (hunk 2 above)
Comment (mayabrandi) [reply tests/EPPs/test_make_kapa_csv.py:22]: Don't agree on these once. Ill make fixures for artifacts that can and will be used several times. Here i think its better to create the artifact within the test

File: tests/conftest.py (hunk 6 above)
Comment (mayabrandi) [reply tests/conftest.py:16]: Absolutely. 👍

**Comment by mayabrandi** [comment]
Thanks for great feedback @moonso @Mropat! I'm done now. You want to approve? Or you want me to do additional changes?

**Review by moonso — APPROVED** [review]
👍 Did it not work to have server in a fixture?

**Comment by Mropat** [comment]
Did not make sense to have server as fixture, I think. It returns nothing and has no type, and fixtures cannot be dynamic, so in the end one has to make multiple server fixtures (but server function can spin up server with any given file path as argument)

**Comment by mayabrandi** [comment]
I made server fixtures fo the servers that will be reused several times but not for the specific once used on or two times. @moonso
